packages/edit-site/src/components/global-styles/screen-revisions/revisions-buttons.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __, _x, sprintf } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import { dateI18n, getDate, humanTimeDiff } from '@wordpress/date';
+
+/**
+ * Returns a button label for the revision.
+ *
+ * @param {Object} revision A revision object.
+ * @return {string} Translated label.
+ */
+function getRevisionLabel( revision ) {
+	const authorDisplayName = revision?.author?.name;
+	const isUnsaved = 'unsaved' === revision?.id;
+
+	if ( isUnsaved ) {
+		return sprintf(
+			/* translators: %(name)s author display name */
+			__( 'Unsaved changes by %(name)s' ),
+			{
+				name: authorDisplayName,
+			}
+		);
+	}
+	const date = getDate( revision?.modified );
+	const formattedDate = dateI18n(
+		// translators: a compact version of the revision's modified date.
+		_x( 'j M @ H:i', 'formatted version of revision last modified date' ),
```

Comment:
I'd use `at` instead of `@` since this is going into an `aria-label` and screen readers / translators might struggle with the symbol.

Also, the default locale (which is then translated) is `en_US` so we should use the American way of writing dates and times here.

```suggestion
		_x( 'M j g:i a', 'formatted version of revision last modified date' ),
```

Reply:
We could also just use `wp.date.getSettings().formats.datetimeAbbreviated` and not renitent the wheel. That way it's somebody else's problem 😀

Reply:
> i'd use at instead of @ since this is going into an aria-label and screen readers / translators might struggle with the symbol.

It's the format that we use on the revisions pages, that's why I went with it.

<img width="342" alt="Screenshot 2023-05-05 at 12 28 01 pm" src="https://user-images.githubusercontent.com/6458278/236366104-d59cdaf8-5411-4394-ab96-49ba4b7bff7d.png">

> We could also just use wp.date.getSettings().formats.datetimeAbbreviated and not reinvent the wheel. That way it's somebody else's problem 

That sounds like a much better idea 😄